src/main/java/step4/Cars.java

```diff
@@ -0,0 +1,47 @@
+package step4;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Cars {
+
+    private static final String RANKED_STRING_DELIMITER = ",";
+
+    private List<Car> cars = new ArrayList<>();
+
+    public Cars(List<Car> cars) {
+        this.cars = cars;
+    }
+
+    public void setupCars(String[] carsNames, CarMoveStrategy carMoveStrategy) {
+        Arrays.stream(carsNames).forEachOrdered(name ->
+                cars.add(new Car(name, 0, carMoveStrategy)));
+    }
+
+    public String getWinners() {
+
+        Optional<Car> max = cars.stream().max((o1, o2) -> (o1.getLocation() >= o2.getLocation()) ? 1 : -1);
+        Stream<Car> ranked = cars.stream().filter(o -> o.getLocation() == max.get().getLocation());
+        String winners = ranked.map(Car::getCarName).collect(Collectors.joining(RANKED_STRING_DELIMITER));
+
+        return winners;
+    }
+
+    public void move() {
+        cars.stream().forEach(car -> {
```

Comment:
```suggestion
        cars.forEach(car -> {
```
을 사용하셔도 될 것 같네요:)
Stream.forEach와 Collection.forEach의 차이점에 관해서도 한번 참고해주세요:)

https://stackoverflow.com/questions/23218874/what-is-difference-between-collection-stream-foreach-and-collection-foreach
https://www.baeldung.com/java-collection-stream-foreach

Reply:
문서로 공부 많이 되었네요. 스트림은  병렬 처리라서 막연히 더 빠르겠거니 했는데 꼭 그렇지도 않은거네요 하나 더 배웠습니다. ^^*